asm/instruction.go

```diff
@@ -584,7 +584,7 @@ func (insns Instructions) FixupReferences() error {
 			// no fixup needed
 			continue
 		}
-		return fmt.Errorf("%s at insn %d: symbol %q: %w", ins.OpCode, i, ins.Reference, ErrUnsatisfiedReference)
+		return fmt.Errorf("%s at insn %d: symbol %q: %w", ins.OpCode, i, ins.Reference, ErrUnsatisfiedProgramReference)
```

Comment:
I think this came about because I keep misreading the switch & loop. I'll push something that makes it a bit easier to read.

Reply:
See https://github.com/cilium/ebpf/pull/571/commits/3880052a663a028fedeb5d4f3a0697201d629d97

Reply:
Sure, thanks :+1: